eng/Configurations.props

```diff
@@ -38,6 +37,10 @@
   </PropertyGroup>
 
   <PropertyGroup>
+    <RuntimeRID Condition="'$(RuntimeRID)' == '' and '$(MSBuildRuntimeType)' == 'core'">$([System.Runtime.InteropServices.RuntimeInformation]::RuntimeIdentifier)</RuntimeRID>
```

Comment:
Is `RuntimeRID` a good name for this? We use `Runtime` in `RuntimeConfiguration` to mean either "CoreCLR" or "Mono" as the "Runtime".

(Also, it is a little redundant since "RID" stands for "Runtime Identifier".)

In core-setup we used the term `HostMachineRid`, which meant "the RID of the machine hosting the build". See the description of [https://github.com/dotnet/core-setup/pull/3266](https://github.com/dotnet/core-setup/pull/3266) for the reasoning. It might make sense to use that term here.

Reply:
We discussed this in the past and agreed on the following naming convention: https://github.com/dotnet/runtime/pull/32833#discussion_r384631816.

> (Also, it is a little redundant since "RID" stands for "Runtime Identifier".)

Then probably RuntimeID would be a better fit.

Reply:
I believe that the official SDK uses fully spelled `RuntimeIdentifier`. Should we match this?

How is this different from the `RuntimeIdentifier` property from the official SDK?

> the RID of the machine hosting the build

Should it be the RID of the machine hosting the build, or should it be the RID of the target platform? I would expect it to be the latter.

Reply:
> I believe that the official SDK uses fully spelled RuntimeIdentifier. Should we match this?
How is this different from the RuntimeIdentifier property from the official SDK?

@ericstj should have more context on the sdk's `RuntimeIdentifier` property.

Reply:
> Should it be the RID of the machine hosting the build, or should it be the RID of the target platform?

This was covered in https://github.com/dotnet/core-setup/pull/3266 as well. There are cases for needing both (and the `toolset` RID as well), which is why we split into 3 properties:

- What is the RID of the current machine? $(HostMachineRid)
- What is the RID of the toolset we are using to build? core-setup doesn't have this scenario yet, but corefx calls this $(ToolRuntimeRID)
- What is the RID that the current build is producing outputs for? $(OutputRid)

Reply:
Here's a concrete reason why `the RID of the machine hosting the build` != `the RID of the target platform` - when building "portable" on a Linux machine, the `the RID of the target platform` will be `linux-x64`. But we need to know if we should build `.deb` packages or `.rpm` packages. To make this decision, we look at the `the RID of the machine hosting the build` to see if it is debian based or not.

Reply:
Is check against a hardcoded list really the proper way to check for debian-based systems? Number of places on the internet suggest to check `/etc/debian_version` for this.

Reply:
If there's a better way to implement that one example, we should do it.

But there are other usages of `the RID of the machine hosting the build` - basically anywhere we are reading `RuntimeOS` today. Another one is to default `$(ToolRuntimeRID)` - 

https://github.com/dotnet/runtime/blob/817ebcaeb617813f98650e78c14e76bbd8b733b0/src/libraries/Directory.Build.props#L90-L91
https://github.com/dotnet/runtime/blob/817ebcaeb617813f98650e78c14e76bbd8b733b0/src/libraries/Directory.Build.props#L107

Reply:
For the name of this property, we already use the name `Host` in `HostArch`:

https://github.com/dotnet/runtime/blob/817ebcaeb617813f98650e78c14e76bbd8b733b0/src/libraries/Directory.Build.props#L34-L35

Calling this new property `HostRid` or `HostRuntimeIdentifier` would align with that.

Reply:
>  there are other usages of the RID of the machine hosting the build - basically anywhere we are reading RuntimeOS today

RuntimeOS means machine that we are targeting, not the machine hosting the build.

Reply:
> RuntimeOS means machine that we are targeting

In a "portable" build, we are targeting `linux`, right? If that is true, then this code in the PR:

https://github.com/dotnet/runtime/pull/35538/files#diff-9f3c793773fc53c31ea09d056e98a4eeR294-R296

is not correct.

> RuntimeOS means machine that we are targeting, not the machine hosting the build.

This is exactly why I'm asking for a different name for this new property. `RuntimeOS` meaning "The OS we are targeting" and `RuntimeRID` meaning "The RID of the machine we are running on" is confusing. Also having to explain to new people that "RuntimeOS means machine that we are targeting, not the machine hosting the build." suggests it is not a very descriptive name.

Reply:
Makes sense, I will rename the property to `HostRuntimeIdentifier`.

Reply:
I logged https://github.com/dotnet/runtime/issues/35607 to update the installer build with a better way to check if running on a Debian-based OS.